.github/workflows/R-CMD-check.yml

```diff
@@ -75,37 +49,35 @@ jobs:
       - name: Install apt packages (Linux)
         if: runner.os == 'Linux'
         run: |
-          sudo apt-get install \
-            texlive-latex-base texlive-fonts-recommended libcurl4-openssl-dev
+            sudo apt-get update
+            sudo apt-get install -y texlive-latex-base texlive-fonts-recommended libcurl4-openssl-dev libglpk40 
 
-      - name: Set up R dependencies (Windows)
-        if: runner.os == 'Windows'
-        uses: r-lib/actions/setup-r-dependencies@v2
-        with:
-          extra-packages: |
-            ms609/TreeDistData
-          needs: |
-            check
-            coverage
-            
-      - name: Set up R dependencies (Non-Windows)
-        if: runner.os != 'Windows'
-        uses: r-lib/actions/setup-r-dependencies@v2
-        with:
-          extra-packages: |
-            ms609/TreeDistData
-            phangorn=?ignore-before-r=4.1.0
-            RcppCore/Rcpp
-          needs: |
-            check
+      - name: Install devtools and remotes
+        run: Rscript -e 'install.packages(c("devtools", "remotes"))'
+
+      - name: Install pak
+        run: Rscript -e 'install.packages("pak")'
+
+      - name: Install covr
+        run: Rscript -e 'install.packages("covr")'
+
+      - name: Install latest Rcpp from GitHub
+        run: Rscript -e 'remotes::install_github("RcppCore/Rcpp")'
+
+      - name: Install TreeDistData from GitHub
+        run: Rscript -e 'devtools::install_github("ms609/TreeDistData")'
+
+      - name: Install dependencies
+        run: Rscript -e 'pak::pkg_install(c("fGarch", "univariateML", "kdensity"))'
+
+      - name: Install remaining dependencies
+        run: Rscript -e 'pak::pkg_install(c("TreeDist", "phangorn", "Rcpp"))'
 
       - name: Set up pandoc
         uses: r-lib/actions/setup-pandoc@v2
-        
+
       - name: Check package
         uses: r-lib/actions/check-r-package@v2
 
       - name: Code coverage
-        if: runner.os == 'Windows'
-        run: covr::codecov()
-        shell: Rscript {0}
+        run: Rscript -e 'covr::codecov()'
```

Comment:
Running this under all OS will result in duplication; unclear what is gained here

Reply:
You are right. I think it is not something new. It used to be that way. [https://github.com/ms609/TreeDist/blob/c57c9cc46d74697698ac0eeaa3e32b6e67e016c9/.github/workflows/R-CMD-check.yml](https://github.com/ms609/TreeDist/blob/c57c9cc46d74697698ac0eeaa3e32b6e67e016c9/.github/workflows/R-CMD-check.yml)
Actually it will fail on windows before repeating cause it is not a proper shell command like this `Rscript -e 'covr::codecov()'`